tests/supply_chain/test_env_reset.py

```diff
@@ -0,0 +1,474 @@
+        self.assertEqual(10, seller_unit._demand)
+        self.assertEqual(10, seller_unit._total_sold)
+
+        expect_tick = 12
+
+        # Save the env.metric of each tick into env_metric_1
+        env_metric_1: Dict[int, dict] = defaultdict(dict)
+
+        # Store the information about the snapshot seller unit of each tick in states_1
+        states_1: Dict[int, list] = defaultdict(list)
+        for i in range(expect_tick):
+            env.step(None)
+            env_metric_1[i] = env.metrics
+            states_1[i] = seller_nodes[i:seller_node_index:features].flatten().astype(np.int)
+
+        # ################# Test whether reset updates the seller unit completely ################
+        env.reset()
+        env.step(None)
+
+        # snapshot should reset after env.reset().
+        states = seller_nodes[1:seller_node_index:features].flatten().astype(np.int)
+        self.assertEqual([0, 0, 0, 0, 0, 0, 0, 0], list(states))
+
+        expect_tick = 12
+
+        # Save the env.metric of each tick into env_metric_2
+        env_metric_2: Dict[int, dict] = defaultdict(dict)
+
+        # Store the information about the snapshot seller unit of each tick in states_2
+        states_2: Dict[int, list] = defaultdict(list)
+
+        for i in range(expect_tick):
+            env.step(None)
+            env_metric_2[i] = env.metrics
+            states_2[i] = seller_nodes[i:seller_node_index:features].flatten().astype(np.int)
+
+        for i in range(expect_tick):
+            self.assertEqual(list(states_1[i]), list(states_2[i]))
+            self.assertEqual(list(env_metric_1[i].values()), list(env_metric_2[i].values()))
+
+    def test_storage_unit_reset(self) -> None:
+        """Facility with single SKU. -- Supplier_SKU3"""
+        env = build_env("case_01", 100)
+        be = env.business_engine
+        assert isinstance(be, SupplyChainBusinessEngine)
+
+        env.step(None)
+
+        supplier_3: FacilityBase = be.world._get_facility_by_name("Supplier_SKU3")
+
+        storage_unit: StorageUnit = supplier_3.storage
+        storage_node_index = storage_unit.data_model_index
+        storage_nodes = env.snapshot_list["storage"]
+        features = ("id", "facility_id",)
+
+        # ##################################### Before reset #####################################
+
+        expect_tick = 10
+
+        # Save the env.metric of each tick into env_metric_1
+        env_metric_1: Dict[int, dict] = defaultdict(dict)
+
+        # Store the information about the snapshot storage unit of each tick in states_1
+        states_1: Dict[int, list] = defaultdict(list)
+        for i in range(expect_tick):
+            env.step(None)
+            env_metric_1[i] = env.metrics
+            states_1[i] = list(storage_nodes[i:storage_node_index:features].flatten().astype(np.int))
+            states_1[i].append(storage_nodes[i:storage_node_index:"product_id_list"].flatten().astype(np.int).sum())
```

Comment:
why sum them up?

Reply:
I understand it this way, because many attributes of the snapshot inch of storage are NodeAttributeAccessor objects, which are of ndarray type after being taken out from the snapshot. I was afraid that it would be difficult to compare, so I added sum().